encord_agents/tasks/runner.py

```diff
@@ -460,6 +468,8 @@ def {fn_name}(...):
                     batch_lrs: list[LabelRowV2 | None] = []
 
                     tasks = list(stage.get_tasks())
+                    if max_tasks_per_stage:
+                        tasks = tasks[:max_tasks_per_stage]
```

Comment:
```suggestion
                    tasks = list(stage.get_tasks())[:max_tasks_per_stage]
```

Example:
```python
> a = [1, 2, 3]
> a[:None]
[1, 2, 3]
> a[:2]
[1, 2]
> a[:500]
[1, 2, 3]
```

If you index with None, you get the full list. 
If number higher than len(list) you get full list.

So with validator (>=1 | None) we should be good.

Reply:
I dislike this personally cause it's not as explicit but happy to make the change

Reply:
Already made the change